packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -174,7 +174,11 @@ class _RenderCompositionCallback extends RenderProxyBox {
 /// [TextInputFormatter]s instead for as-you-type text modification.
 ///
 /// If both the [text] and [selection] properties need to be changed, set the
-/// controller's [value] instead.
+/// controller's [value] instead. In some situations, only setting the controller's
+/// [text] property seems sufficient, but is not. For example, to clear
+/// a [TextField], consider using [TextEditingController.clear]
+/// instead of setting [TextEditingController.text] to the empty string
+/// to also reset the [selection].
```

Comment:
I'm thinking simply add this to the end of the original paragraph: "Setting [text] will clear the selection and composing range."